api/pkg/cli/cmd/check_upgrade/check_uprade.go

```diff
@@ -0,0 +1,216 @@
+// Copyright © 2021 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package check_upgrade
+
+import (
+	"strings"
+	"text/template"
+
+	"github.com/spf13/cobra"
+	"github.com/tektoncd/hub/api/pkg/cli/app"
+	"github.com/tektoncd/hub/api/pkg/cli/formatter"
+	"github.com/tektoncd/hub/api/pkg/cli/hub"
+	"github.com/tektoncd/hub/api/pkg/cli/installer"
+	"github.com/tektoncd/hub/api/pkg/cli/kube"
+	"github.com/tektoncd/hub/api/pkg/cli/printer"
+)
+
+const upgradeTemplate = `{{decorate "underline bold" "Hub Resources\n" }}
+{{- $len := len .HubResources }}{{- if ne $len 0 }}
+NAME	CATALOG	CURRENT_VERSION	NEWER_VERSION
+{{- range $el := .HubResources }}
+{{ $el.Name }}	{{ $el.Catalog }}	{{ $el.CurrentVersion }}	{{ $el.LatestVersion }}
+{{- end }}
+{{- else }}
+No {{ .Kind }} for upgrade
+{{- end }}
+{{ $len := len .NonHubResources }}{{ if ne $len 0 }}
+{{ decorate "underline bold" "Skipped Resources\n" }}
+NAME
+{{- range $el := .NonHubResources }}
+{{ icon "bullet" }}{{ $el }}
+{{- end }}
+{{- end }}
+{{ if .IsPipelineVersionUnknown }}
+{{ decorate "bold" "Pipelines version unknown. Upgrade the resources at your own risk." }}
+{{- end }}
+`
+
+var (
+	funcMap = template.FuncMap{
+		"icon":     formatter.Icon,
+		"decorate": formatter.DecorateAttr,
+	}
+	tmpl = template.Must(template.New("Check Upgrade").Funcs(funcMap).Parse(upgradeTemplate))
+)
+
+type hubRes struct {
+	Name           string
+	Catalog        string
+	CurrentVersion string
+	LatestVersion  string
+}
+
+type templateData struct {
+	Kind                     string
+	NonHubResources          []string
+	HubResources             []hubRes
+	IsPipelineVersionUnknown bool
+}
+
+const (
+	versionLabel = "app.kubernetes.io/version"
+	hubLabel     = "hub.tekton.dev/catalog"
+)
+
+type options struct {
+	cli     app.CLI
+	kind    string
+	version string
+	args    []string
+	kc      kube.Config
+	cs      kube.ClientSet
+}
+
+var cmdExamples string = `
+Check for Upgrades of %S installed via Tekton Hub CLI:
+
+	tkn hub check-upgrades %s
+
+The above command will check for upgrades of %Ss installed via Tekton Hub CLI
+and will skip the %Ss which are not installed by Tekton Hub CLI.
+`
```

Comment:
should we document it if we don't find pipelines version it shows latest otherwise latest compatible?

Reply:
makes sense. Will add that.